sys/include/net/ng_netif/hdr.h

```diff
@@ -124,15 +127,58 @@ inline uint8_t *ng_netif_hdr_get_dst_addr(ng_netif_hdr_t *hdr)
  * @param[in] addr_len      *addr* length
  */
 inline void ng_netif_hdr_set_dst_addr(ng_netif_hdr_t *hdr, uint8_t *addr,
-                                  uint8_t addr_len)
+                                      uint8_t addr_len)
 {
     if (addr_len != hdr->dst_l2addr_len) {
         return;
     }
+
     memcpy(((uint8_t *)hdr) + sizeof(ng_netif_hdr_t) + hdr->dst_l2addr_len,
            addr, addr_len);
 }
 
+/**
+ * @brief   Builds a generic network interface header for sending and
+ *          adds it to the packet buffer.
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * @param[in] payload   Payload for the packet.
+ * @param[in] src       Source address for the header. Can be NULL if not
+ *                      known or required.
+ * @param[in] src_len   Length of @p src. Can be 0 if not known or required.
+ * @param[in] dest      Destination address for the header. Can be NULL if not
+ *                      known or required.
+ * @param[in] dest_len  Length of @p dest. Can be 0 if not known or required.
+ *
+ * @return  The generic network layer header on success.
+ * @return  NULL on error.
+ */
+static inline ng_pktsnip_t *ng_netif_hdr_build(ng_pktsnip_t *payload,
```

Comment:
Did you read the doc?

>+ * @brief   Builds a generic network interface header for **sending** and
+ *          adds it to the packet buffer.

The reason why it looks like for receiving is because I had it reversed :D